### PR TITLE
Decouple export data progress from reporting it on terminal screen

### PR DESCRIPTION
**Pull request by rahulb-yb**
The issue:
- Progress bars would be updated side-by-side with maintaining a structure which keeps a track of the data progress across tables
- Disabling progress bars using `--disable-pb` would disable this logic of maintaining the structure which stores the exported data progress; we would get an incorrect count of 0 rows even though the export would successfully go through.
- A more recent issue: The logic to skip for when partitions are empty is coupled with our progress bar logic (we need to exit all progress bars in such scenarios); this logic gets skipped entirely and export data remains hanging when we disable progress bars, and some partitions are empty.

The solution:
Create an interface which closely resembles how we use the `mpb` package. When progress bars are enabled, the interface acts as a wrapper while adding more readability. When they are disabled, the interface mimics the barebones of what yb-voyager needs from the mpb package, without ever tying itself to a progress container (which in turn makes the progress bars "invisible")

Future tasks:
- Enable partition/subpartition automation framework for MySQL
- Enhance export data status command

### CODE REVIEW

File: yb-voyager/src/pbreporter/disablePBReporter.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	}
+	if triggerComplete && !pbr.IsCompleted {
+		pbr.IsCompleted = true
+		pbr.CurrentRows = pbr.TotalRows
+	}
+
+}
+func (pbr DisablePBReporter) SetExportedRowCount(exportedRowCount int64) {
+	if exportedRowCount < 0 {
+		utils.ErrExit("cannot maintain negative export row count in PB")
```
Comment (rahulb-yb) [inline yb-voyager/src/pbreporter/disablePBReporter.go:31]: export-exported

File: yb-voyager/src/pbreporter/exportProgressReporter.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+type ExportProgressReporter interface { // Bare minimum required to simulate mpb.bar for exportDataStatus
+	SetTotalRowCount(totalRowCount int64, triggerComplete bool)
+	SetExportedRowCount(exportedRowCount int64)
+	IsComplete() bool
+}
+
+func NewExportPB(progressContainer *mpb.Progress, tableName string, disablePb bool) ExportProgressReporter {
+	if disablePb {
+		return newDisablePBReporter()
+	} else {
```
Comment (rahulb-yb) [inline yb-voyager/src/pbreporter/exportProgressReporter.go:14]: remove else condition

File: yb-voyager/src/pbreporter/disablePBReporter.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+type DisablePBReporter struct { // Each individual goroutine has the context of the table corresponding to the PBReporter, no need to add to struct.
+	TotalRows       int64
+	CurrentRows     int64
+	IsCompleted     bool
+	TriggerComplete bool
+}
+
+func newDisablePBReporter() *DisablePBReporter {
+	return &DisablePBReporter{TotalRows: int64(0), CurrentRows: int64(0), IsCompleted: false, TriggerComplete: false}
```
Comment (amit-yb) [inline yb-voyager/src/pbreporter/disablePBReporter.go:13]: Nit: You can simply use the default values of each type:
```
return &DisablePBReporter{}
```

File: yb-voyager/src/pbreporter/disablePBReporter.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	} else {
+		pbr.TotalRows = totalRowCount
+	}
+	if triggerComplete && !pbr.IsCompleted {
+		pbr.IsCompleted = true
+		pbr.CurrentRows = pbr.TotalRows
+	}
+
+}
+func (pbr *DisablePBReporter) SetExportedRowCount(exportedRowCount int64) {
```
Comment (amit-yb) [inline yb-voyager/src/pbreporter/disablePBReporter.go:29]: One empty line between each function definition.
Pass your code through `go fmt`. Or set your IDE to run `go fmt` on each save.

**Review by amit-yb — APPROVED** [review]
LGTM.

**Pull request merged**